adapters/openx/openx.go

```diff
@@ -143,11 +143,13 @@ func preprocess(imp *openrtb.Imp, reqExt *openxReqExt) error {
 	}
 
 	if imp.Video != nil {
+		videoCopy := *imp.Video
```

Comment:
@laurb9 I missed this in the review of https://github.com/prebid/prebid-server/pull/1290. Video is shared memory and should be copied before mutation.

Reply:
Thanks. I thought this was safe because the adapter makes a copy of the entire request (unlike others), but I should have noticed the pointer.

By the way, do you use a tool to find these problems ? I'm just starting in Go and this could be useful to add to my toolbox.

Reply:
IMHO, Go doesn't provide the necessary syntax to express memory safety in code as other languages do. The best tool we have to find these kinds of issues is the [race detector](https://golang.org/doc/articles/race_detector.html). However, that tool isn't a static analyzer. It's limited to checking code that's running during a test (which is why we push for race detector test cases so much during code reviews). If you find a good static memory safety analyzer, please let us know. :)